src/sdk/sts_sdk.js

```diff
@@ -0,0 +1,100 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const cloud_utils = require('../util/cloud_utils');
+const dbg = require('../util/debug_module')(__filename);
+const LRUCache = require('../util/lru_cache');
+const { RpcError } = require('../rpc');
+const signature_utils = require('../util/signature_utils');
+const db_client = require('../util/db_client').instance();
+
+// DO we need that cache here? 
+const account_cache = new LRUCache({
+    name: 'AccountCache',
+    // TODO: Decide on a time that we want to invalidate
+    expiry_ms: 10 * 60 * 1000,
+    make_key: ({ access_key }) => access_key,
+    load: async ({ rpc_client, access_key }) => rpc_client.account.read_account_by_access_key({ access_key }),
+});
+
+
+class StsSDK {
+
+    constructor(rpc_client, internal_rpc_client) {
+        this.rpc_client = rpc_client;
+        this.internal_rpc_client = internal_rpc_client;
+        this.requesting_account = undefined;
+    }
+
+    set_auth_token(auth_token) {
+        this.rpc_client.options.auth_token = auth_token;
+    }
+
+    get_auth_token() {
+        return this.rpc_client.options.auth_token;
+    }
+
+    async get_assumed_role(req) {
+        dbg.log0('sts_sdk.get_assumed_role params', req.params);
+        // arn:aws:sts::account_id:role/role_name
+
+
+        const role_name_idx = req.params.role_arn.lastIndexOf('/') + 1;
+        const role_name = req.params.role_arn.slice(role_name_idx);
+        const account_id_str = req.params.role_arn.split(':')[4];
+        let account_id;
+        try {
+            account_id = db_client.parse_object_id(account_id_str);
+        } catch (err) {
+            throw new RpcError('NO_SUCH_ACCOUNT', 'No such account with id: ' + account_id_str);
+        }
+        if (!account_id) throw new RpcError('NO_SUCH_ACCOUNT', 'No such account with id: ' + account_id);
+
+        let account = await this.rpc_client.account.read_account_by_id({ account_id });
```

Comment:
why is this not using the accounts_cache?

Reply:
the key of account cache is access key and not account id, but after re-discussing it with @jeniawhite, I changed the account identifier when calling assume role to access_key since the account id is internal to the database, so I changed this line to use the account cache.